### ISSUE TITLE
Run tests from executable jar

### ISSUE THREAD

**Issue by sergiuciudin** [serenity-bdd/serenity-jbehave#205]
Is there a way to run tests from main?

**Comment by wakaleo** [comment]
Yes, but not easily - it would involve some custom code and custom build scripting.

**Comment by sergiuciudin** [comment]
@wakaleo thank you for the fast reply :)

Can you share please the idea?

I'm trying now to do something like this:

`
public class AcceptanceTest extends SerenityStories {
    @Managed(driver = "chrome", uniqueSession = true, clearCookies = BeforeEachTest)
    WebDriver driver;

    @Override
    @Test
    public void run() throws Throwable {
        super.run();
    }
}
`

`public static void main(String[] args) {
        JUnitCore core = new JUnitCore();
        core.run(AcceptanceTestS.class);
    }`

**Comment by franciscoguemes** [comment]
Could you provide an example or some guidelines on how to run the tests with serenity from a *.jar file?

**Comment by wakaleo** [comment]
It really isn't trivial - I implemented something like this for a client a few years back for pure JUnit and it requires a fairly deep knowledge of JUnit and Serenity internals. I wouldn't know for JBehave.

**Comment by franciscoguemes** [comment]
And what about Cucumber-JVM ?

**Comment by Constantinul1** [comment]
Hello, any details on how to do this? It's becoming more and more necessary apparently, to the point where we're forced to drop the framework solely because it cannot be done. A shame, really.

**Comment by wakaleo** [comment]
What problem are you trying to solve?

**Comment by Constantinul1** [comment]
Ideally, having the ability to run individual tests from a packaged jar.

**Comment by wakaleo** [comment]
And what will this allow you to do? Test suites are typically run as part of a build process, so creating an executable jar from a Java project that includes both the tests and the application code, and a runner, really isn't simple. It's not Serenity, it's the whole Java development approach.

**Comment by Constantinul1** [comment]
It would be just the packaged serenity project, not the application code as well. This way tests can be ran in parallel on AWS for example, or any number of other variations. The setup demands that the tests are ran in the deploy phase.

**Comment by wakaleo** [comment]
You could do this by putting all the tests in src/main/java and creating a main() method that triggers the JUnit runner classes. But wouldn't it be easier just to use Docker?

**Comment by Constantinul1** [comment]
Could be, I'll give it a try. Thank you for your answer.

**Comment by vulkman** [comment]
@wakaleo Got it to work, only problem: I don't get an index.html in the results, probably because the step that's usually done by serenity:aggregate is missing. Could you tell me how to run that directly in my main()?

**Comment by msamarba** [comment]
Hello @vulkman can you please tell me how you got it working ? I currently use the maven-assembly-plugin and an assembly.xml to create that jar file. However running it locally and in Jenkins gives me nullpointerexceptions. It seems it can't see Serenity in the jar file.
